version: '3'
services:
  file-converter-beat:
    build: ./../appNube_Miso
    environment:
      - DATABASE_URI=mysql://root:})^M^JeM?H0+=e(-@35.233.184.37/file-converter
      - RABBIT_HOST=rabbit
    command:
      - celery
      - -A
      - tasks.tasks
      - beat
      - --loglevel=info
    volumes:
      - ./files:/app/files
    depends_on:
      - rabbit
  file-converter-worker:
    build: ./../appNube_Miso
    environment:
      - DATABASE_URI=mysql://root:})^M^JeM?H0+=e(-@35.233.184.37/file-converter
      - RABBIT_HOST=rabbit
    command:
      - celery
      - -A
      - tasks.tasks
      - worker
      - --loglevel=info
      - -P
      - threads
    volumes:
      - ./files:/app/files
    depends_on:
      - file-converter-beat
      - rabbit
  file-converter-service:
    build: ./../appNube_Miso
    environment:
      - DATABASE_URI=mysql://root:})^M^JeM?H0+=e(-@35.233.184.37/file-converter
    command:
      - python
      - app.py
    volumes:
      - ./files:/app/files
    ports:
      - "5000:5000"
  rabbit:
    image: rabbitmq:3.7.5-management
    ports:
      - "15672:15672"
      - "5672:5672"
      - "4369:4369"
      - "5671:5671"
    command: sh -c "rabbitmq-plugins enable rabbitmq_management;  echo '[{rabbit, [{loopback_users, []}]}].' > /etc/rabbitmq/rabbitmq.config; rabbitmq-server"